cvat-core/src/frames.ts

```diff
@@ -208,7 +208,23 @@ export class FramesMetaData {
             // it may be a videofile or one image
             framesInfo = frameNumbers.map(() => initialData.frames[0]);
         } else {
-            framesInfo = initialData.frames;
+            framesInfo = (() => {
+                if (initialData.frames.length > 1 &&
```

Comment:
there is condition `initialData.frames.length === 1` above

Reply:
The solution with frame names looks like a dirty hack
What to you think about this one?

```ts
            if (this.includedFrames?.length) {
                // only gt jobs have this.includedFrame, so this condition works only for them
                const framesIndexes = new Set(this.includedFrames.map((dataFrameNumber: number) => {
                    return Math.floor((dataFrameNumber - this.startFrame) / this.frameStep);
                }));
                framesInfo = initialData.frames.filter((_, idx) => framesIndexes.has(idx))
            } else {
                framesInfo = initialData.frames;
            }
```

Reply:
> The solution with frame names looks like a dirty hack

Why it looks this way, if it's a constant defined by the server specifically for this?

> What to you think about this one?

The formula part `dataFrameNumber - this.startFrame` is not correct in cases with nonzero start frame and frame step. In meta `start_frame`, `stop_frame`, and `included_frames` contain input data frame numbers (absolute ones). The correct way would be to obtain the original data frame number, then find the corresponding `included_frames` index.

Reply:
> Why it looks this way, if it's a constant defined by the server specifically for this?

Because as you mentioned, user may have files named `placeholder.jpg`, what potentially may lead to bugs
We should not consider such solutions

Reply:
> The formula part dataFrameNumber - this.startFrame is not correct in cases with nonzero start frame and frame step

I tested it in this case, and it seems to work in cases with non-zero start frame and frame step !== 1. Not really understand from your explanation why it should not. 

> In meta `start_frame`, `stop_frame`, and `included_frames` contain input data frame numbers (absolute ones)

Yes. We want to map absolute frame number to segment frame number, using the formula above. 
Then, as for GT jobs meta returns info about all frames in the task, from 1st one, we do assumption, that frame number == frame index.

Reply:
Ok, looks good to me. Added some explanatory comments why it works.